From f3e45604d2c94715068ff33dba88a7b831dc1a37 Mon Sep 17 00:00:00 2001
From: aobaharuki2005 <tranbaohnth@outlook.com>
Date: Fri, 17 Apr 2026 08:08:32 +0700
Subject: [PATCH 20/26] tools_patchset

---
 tools/profiler/core/platform.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/tools/profiler/core/platform.cpp b/tools/profiler/core/platform.cpp
index dcb11707482..b52fc6b053b 100644
--- a/tools/profiler/core/platform.cpp
+++ b/tools/profiler/core/platform.cpp
@@ -7699,12 +7699,19 @@ void profiler_mark_thread_awake() {
       currentPriority = threadInfo.Priority;
     }
   }
+#endif
+#if defined(GP_OS_darwin)
+  size_t qos_self_retval;
+  if(__builtin_available(macOS 10.10, *)) 
+    qos_self_retval = qos_class_self();
+  else
+    qos_self_retval = QOS_CLASS_UNSPECIFIED;
 #endif
   PROFILER_MARKER("Awake", OTHER, MarkerTiming::IntervalStart(), CPUAwakeMarker,
                   0 /* CPU time */, cpuId
 #if defined(GP_OS_darwin)
                   ,
-                  qos_class_self()
+                  qos_self_retval 
 #endif
 #if defined(GP_OS_windows)
                       ,
-- 
2.53.0

