From 977d6e489fdf141f880ff6137dded37e58c02fb3 Mon Sep 17 00:00:00 2001
From: aobaharuki2005 <tranbaohnth@outlook.com>
Date: Sun, 19 Apr 2026 22:31:39 +0700
Subject: [PATCH 019/246] merge_dom/media/platforms/apple/AppleDecoderModule.h

---
 dom/media/platforms/apple/AppleDecoderModule.h | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/dom/media/platforms/apple/AppleDecoderModule.h b/dom/media/platforms/apple/AppleDecoderModule.h
index 63e959225d..2510c7114f 100644
--- a/dom/media/platforms/apple/AppleDecoderModule.h
+++ b/dom/media/platforms/apple/AppleDecoderModule.h
@@ -7,7 +7,6 @@
 #ifndef mozilla_AppleDecoderModule_h
 #define mozilla_AppleDecoderModule_h
 
-#include "MediaCodecsSupport.h"
 #include "PlatformDecoderModule.h"
 
 namespace mozilla {
@@ -39,9 +38,13 @@ class AppleDecoderModule : public PlatformDecoderModule {
 
   static void Init();
 
+  static bool sIsCoreMediaAvailable;
+  static bool sIsCoreVideoAvailable;
+  static bool sIsVDAAvailable;
+  static bool sIsVTAvailable;
+
   static constexpr int kCMVideoCodecType_H264{'avc1'};
   static constexpr int kCMVideoCodecType_VP9{'vp09'};
-
  private:
   AppleDecoderModule() = default;
   virtual ~AppleDecoderModule() = default;
-- 
2.53.0

