From 73ab7a70f1895def6da8474a0de32bd191f6d7ba Mon Sep 17 00:00:00 2001
From: aobaharuki2005 <tranbaohnth@outlook.com>
Date: Sun, 19 Apr 2026 22:33:05 +0700
Subject: [PATCH 076/246] merge_layout/generic/nsContainerFrame.cpp

---
 layout/generic/nsContainerFrame.cpp | 39 +++++++++++++----------------
 1 file changed, 18 insertions(+), 21 deletions(-)

diff --git a/layout/generic/nsContainerFrame.cpp b/layout/generic/nsContainerFrame.cpp
index 670e520958..43aebdd16c 100644
--- a/layout/generic/nsContainerFrame.cpp
+++ b/layout/generic/nsContainerFrame.cpp
@@ -746,8 +746,9 @@ void nsContainerFrame::SetSizeConstraints(nsPresContext* aPresContext,
     devMaxSize.height = devMinSize.height;
   }
 
+  nsIWidget* rootWidget = aPresContext->GetNearestWidget();
   DesktopToLayoutDeviceScale constraintsScale(MOZ_WIDGET_INVALID_SCALE);
-  if (nsIWidget* rootWidget = aPresContext->GetNearestWidget()) {
+  if (rootWidget) {
     constraintsScale = rootWidget->GetDesktopToDeviceScale();
   }
 
@@ -756,20 +757,16 @@ void nsContainerFrame::SetSizeConstraints(nsPresContext* aPresContext,
   // The sizes are in inner window sizes, so convert them into outer window
   // sizes. Use a size of (200, 200) as only the difference between the inner
   // and outer size is needed.
-  const LayoutDeviceIntSize sizeDiff =
-      aWidget->NormalSizeModeClientToWindowSizeDifference();
-  if (constraints.mMinSize.width) {
-    constraints.mMinSize.width += sizeDiff.width;
-  }
-  if (constraints.mMinSize.height) {
-    constraints.mMinSize.height += sizeDiff.height;
-  }
-  if (constraints.mMaxSize.width != NS_MAXSIZE) {
-    constraints.mMaxSize.width += sizeDiff.width;
-  }
-  if (constraints.mMaxSize.height != NS_MAXSIZE) {
-    constraints.mMaxSize.height += sizeDiff.height;
-  }
+  LayoutDeviceIntSize windowSize =
+      aWidget->ClientToWindowSize(LayoutDeviceIntSize(200, 200));
+  if (constraints.mMinSize.width)
+    constraints.mMinSize.width += windowSize.width - 200;
+  if (constraints.mMinSize.height)
+    constraints.mMinSize.height += windowSize.height - 200;
+  if (constraints.mMaxSize.width != NS_MAXSIZE)
+    constraints.mMaxSize.width += windowSize.width - 200;
+  if (constraints.mMaxSize.height != NS_MAXSIZE)
+    constraints.mMaxSize.height += windowSize.height - 200;
 
   aWidget->SetSizeConstraints(constraints);
 }
@@ -3067,8 +3064,7 @@ void nsContainerFrame::List(FILE* out, const char* aPrefix,
                             ListFlags aFlags) const {
   nsCString str;
   ListGeneric(str, aPrefix, aFlags);
-  ExtraContainerFrameInfo(str,
-                          aFlags.contains(ListFlag::OnlyListDeterministicInfo));
+  ExtraContainerFrameInfo(str);
 
   // Output the frame name and various fields.
   fprintf_stderr(out, "%s <\n", str.get());
@@ -3120,9 +3116,8 @@ void nsContainerFrame::ListChildLists(FILE* aOut, const char* aPrefix,
 
     // Use nsPrintfCString so that %p don't output prefix "0x". This is
     // consistent with nsIFrame::ListTag().
-    nsCString str{nsPrintfCString("%s%s", aPrefix, ChildListName(listID))};
-    ListPtr(str, aFlags, &GetChildList(listID), "@");
-    str += " <\n";
+    const nsPrintfCString str("%s%s@%p <\n", aPrefix, ChildListName(listID),
+                              &GetChildList(listID));
     fprintf_stderr(aOut, "%s", str.get());
 
     for (nsIFrame* kid : list) {
@@ -3134,6 +3129,8 @@ void nsContainerFrame::ListChildLists(FILE* aOut, const char* aPrefix,
   }
 }
 
-void nsContainerFrame::ExtraContainerFrameInfo(nsACString&, bool) const {}
+void nsContainerFrame::ExtraContainerFrameInfo(nsACString& aTo) const {
+  (void)aTo;
+}
 
 #endif
-- 
2.53.0

