From 41fd75c57b4947833d79465b0902144d9849966d Mon Sep 17 00:00:00 2001
From: aobaharuki2005 <tranbaohnth@outlook.com>
Date: Sun, 19 Apr 2026 22:34:13 +0700
Subject: [PATCH 114/246] merge_security/sandbox/mac/SandboxPolicyUtility.h

---
 security/sandbox/mac/SandboxPolicyUtility.h | 46 ++++++++++++++-------
 1 file changed, 30 insertions(+), 16 deletions(-)

diff --git a/security/sandbox/mac/SandboxPolicyUtility.h b/security/sandbox/mac/SandboxPolicyUtility.h
index b279b47638..fcabbec1ea 100644
--- a/security/sandbox/mac/SandboxPolicyUtility.h
+++ b/security/sandbox/mac/SandboxPolicyUtility.h
@@ -13,7 +13,7 @@ static const char SandboxPolicyUtility[] = R"SANDBOX_LITERAL(
 
   (define should-log (param "SHOULD_LOG"))
   (define app-path (param "APP_PATH"))
-  (define app-binary-path (param "APP_BINARY_PATH"))
+  (define macosVersion (string->number (param "MAC_OS_VERSION")))
   (define crashPort (param "CRASH_PORT"))
   (define isRosettaTranslated (param "IS_ROSETTA_TRANSLATED"))
 
@@ -24,21 +24,31 @@ static const char SandboxPolicyUtility[] = R"SANDBOX_LITERAL(
 
   (moz-deny default)
   ; These are not included in (deny default)
-  (moz-deny process-info*)
-  (moz-deny nvram*)
-  (moz-deny file-map-executable)
+  (if (>= macosVersion 1009)  
+    (moz-deny process-info*))
+  ; This isn't available in some older macOS releases.
+  (if (defined? 'nvram*)
+    (moz-deny nvram*))
+  ; This property requires macOS 10.10+
+  (if (defined? 'file-map-executable)
+    (moz-deny file-map-executable))
 
   ; Needed for things like getpriority()/setpriority()/pthread_setname()
-  (allow process-info-pidinfo process-info-setcontrol (target self))
-
-  (if (string=? isRosettaTranslated "TRUE")
-    (allow file-map-executable (subpath "/private/var/db/oah")))
-
-  (allow file-map-executable file-read*
-    (subpath "/System/Library")
-    (subpath "/usr/lib")
-    (subpath app-path)
-    (subpath app-binary-path))
+  (if (>= macosVersion 1009)  
+  (allow process-info-pidinfo process-info-setcontrol (target self)))
+
+  (if (defined? 'file-map-executable)
+    (begin
+      (if (string=? isRosettaTranslated "TRUE")
+        (allow file-map-executable (subpath "/private/var/db/oah")))
+      (allow file-map-executable file-read*
+        (subpath "/System/Library")
+        (subpath "/usr/lib")
+        (subpath app-path)))
+    (allow file-read*
+      (subpath "/System/Library")
+      (subpath "/usr/lib")
+      (subpath app-path)))
 
   (if (string? crashPort)
     (allow mach-lookup (global-name crashPort)))
@@ -63,8 +73,12 @@ static const char SandboxPolicyUtility[] = R"SANDBOX_LITERAL(
 
 static const char SandboxPolicyUtilityAudioDecoderAppleMediaAddend[] =
     R"SANDBOX_LITERAL(
-  ; For Utility AudioDecoder AppleMedia codecs (bug 1565575)
-  (allow mach-lookup (global-name "com.apple.audio.AudioComponentRegistrar"))
+  ; For Utility AudioDecoder AppleMedia codecs
+  (define macosVersion (string->number (param "MAC_OS_VERSION")))
+  (if (>= macosVersion 1013)
+   (allow mach-lookup
+    ; bug 1565575
+    (global-name "com.apple.audio.AudioComponentRegistrar")))
 )SANDBOX_LITERAL";
 
 }  // namespace mozilla
-- 
2.53.0

