From b88087ad0a55ab81de2fe6aa7df83519db612b6e Mon Sep 17 00:00:00 2001
From: aobaharuki2005 <tranbaohnth@outlook.com>
Date: Sun, 19 Apr 2026 22:35:48 +0700
Subject: [PATCH 170/246] merge_widget/LookAndFeel.h

---
 widget/LookAndFeel.h | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/widget/LookAndFeel.h b/widget/LookAndFeel.h
index 36e2b7ac7b..de3f1c0060 100644
--- a/widget/LookAndFeel.h
+++ b/widget/LookAndFeel.h
@@ -104,6 +104,35 @@ class LookAndFeel {
      * should return NS_ERROR_NOT_IMPLEMENTED when queried for this metric.
      */
     WindowsAccentColorInTitlebar,
+    /*
+     * A Boolean value to determine whether the Mac graphite theme is
+     * being used.
+     *
+     * The value of this metric is not used on other platforms. These platforms
+     * should return NS_ERROR_NOT_IMPLEMENTED when queried for this metric.
+     */
+    MacGraphiteTheme,
+
+    /*
+     * A Boolean value to determine whether the Mac OS X Lion-specific theming
+     * should be used.
+     *
+     * The value of this metric is not used on non-Mac platforms. These
+     * platforms should return NS_ERROR_NOT_IMPLEMENTED when queried for this
+     * metric.
+     */
+    MacLionTheme,
+
+   /*
+    * A Boolean value to determine whether the Mac OS X Yosemite-specific theming
+    * should be used.
+    *
+    * The value of this metric is not used on non-Mac platforms. These
+    * platforms should return NS_ERROR_NOT_IMPLEMENTED when queried for this
+    * metric.
+    */
+
+    MacYosemiteTheme,
 
     /* Whether Windows mica effect is enabled and available */
     WindowsMica,
-- 
2.53.0

