From 686be6b12d8208a3e86e6a121986816707f39bde Mon Sep 17 00:00:00 2001
From: aobaharuki2005 <tranbaohnth@outlook.com>
Date: Sun, 19 Apr 2026 22:35:50 +0700
Subject: [PATCH 172/246] merge_widget/cocoa/GfxInfo.mm

---
 widget/cocoa/GfxInfo.mm | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/widget/cocoa/GfxInfo.mm b/widget/cocoa/GfxInfo.mm
index 8358f6b08b..2173ad15c4 100644
--- a/widget/cocoa/GfxInfo.mm
+++ b/widget/cocoa/GfxInfo.mm
@@ -501,9 +501,6 @@ nsresult GfxInfo::GetFeatureStatusImpl(
         case OperatingSystem::OSX10_5:
         case OperatingSystem::OSX10_6:
         case OperatingSystem::OSX10_7:
-          *aStatus = nsIGfxInfo::FEATURE_BLOCKED_OS_VERSION;
-          aFailureId = "FEATURE_FAILURE_CANVAS_OSX_VERSION";
-          break;
         default:
           *aStatus = nsIGfxInfo::FEATURE_STATUS_OK;
           break;
-- 
2.53.0

