From 8e8449f8e22de2b9b24f3922ad9dab65bedf6d35 Mon Sep 17 00:00:00 2001
From: aobaharuki2005 <tranbaohnth@outlook.com>
Date: Sun, 19 Apr 2026 22:35:51 +0700
Subject: [PATCH 173/246] merge_widget/cocoa/MacThemeGeometryType.h

---
 widget/cocoa/MacThemeGeometryType.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/widget/cocoa/MacThemeGeometryType.h b/widget/cocoa/MacThemeGeometryType.h
index d64580f029..1a442e9975 100644
--- a/widget/cocoa/MacThemeGeometryType.h
+++ b/widget/cocoa/MacThemeGeometryType.h
@@ -9,6 +9,17 @@ enum MacThemeGeometryType {
   eThemeGeometryTypeTitlebar = 1,
   eThemeGeometryTypeSidebar,
   eThemeGeometryTypeWindowButtons,
+  eThemeGeometryTypeFullscreenButton,
+  eThemeGeometryTypeMenu,
+  eThemeGeometryTypeHighlightedMenuItem,
+  eThemeGeometryTypeVibrancyLight,
+  eThemeGeometryTypeVibrancyDark,
+  eThemeGeometryTypeVibrantTitlebarLight,
+  eThemeGeometryTypeVibrantTitlebarDark,
+  eThemeGeometryTypeTooltip,
+  eThemeGeometryTypeSourceList,
+  eThemeGeometryTypeSourceListSelection,
+  eThemeGeometryTypeActiveSourceListSelection
 };
 
 #endif
-- 
2.53.0

