From bbb0299141f266cb11b8bf93d842b7d09b2906ea Mon Sep 17 00:00:00 2001
From: aobaharuki2005 <tranbaohnth@outlook.com>
Date: Sun, 19 Apr 2026 22:36:12 +0700
Subject: [PATCH 182/246] merge_widget/cocoa/TextInputHandler.mm

---
 widget/cocoa/TextInputHandler.mm | 89 ++++++++++++++------------------
 1 file changed, 39 insertions(+), 50 deletions(-)

diff --git a/widget/cocoa/TextInputHandler.mm b/widget/cocoa/TextInputHandler.mm
index 4d64fcc6ed..27195ea747 100644
--- a/widget/cocoa/TextInputHandler.mm
+++ b/widget/cocoa/TextInputHandler.mm
@@ -20,7 +20,8 @@
 #include "mozilla/TextEvents.h"
 #include "mozilla/ToString.h"
 
-#include "nsCocoaWindow.h"
+#include "nsChildView.h"
+#include "nsCocoaFeatures.h"
 #include "nsObjCExceptions.h"
 #include "nsBidiUtils.h"
 #include "nsToolkit.h"
@@ -1789,7 +1790,7 @@ void TextInputHandler::DebugPrintAllKeyboardLayouts() {
  *
  ******************************************************************************/
 
-TextInputHandler::TextInputHandler(nsCocoaWindow* aWidget,
+TextInputHandler::TextInputHandler(nsChildView* aWidget,
                                    NSView<mozView>* aNativeView)
     : IMEInputHandler(aWidget, aNativeView) {
   EnsureToLogAllKeyboardLayoutsAndIMEs();
@@ -1832,7 +1833,7 @@ bool TextInputHandler::HandleKeyDownEvent(NSEvent* aNativeEvent,
     [NSCursor setHiddenUntilMouseMoves:YES];
   }
 
-  RefPtr<nsCocoaWindow> widget(mWidget);
+  RefPtr<nsChildView> widget(mWidget);
 
   KeyEventState* currentKeyEvent = PushKeyEvent(aNativeEvent, aUniqueId);
   AutoKeyEventStateCleaner remover(this);
@@ -2047,7 +2048,7 @@ void TextInputHandler::HandleFlagsChanged(NSEvent* aNativeEvent) {
     return;
   }
 
-  RefPtr<nsCocoaWindow> kungFuDeathGrip(mWidget);
+  RefPtr<nsChildView> kungFuDeathGrip(mWidget);
   mozilla::Unused << kungFuDeathGrip;  // Not referenced within this function
 
   MOZ_LOG_KEY_OR_IME(
@@ -2317,7 +2318,7 @@ void TextInputHandler::HandleFlagsChanged(NSEvent* aNativeEvent) {
                               modifierFlags:modifiers
                                   timestamp:[aNativeEvent timestamp]
                                windowNumber:[aNativeEvent windowNumber]
-                                    context:nil
+                                    context:[aNativeEvent context]
                                  characters:@""
                 charactersIgnoringModifiers:@""
                                   isARepeat:NO
@@ -2568,7 +2569,7 @@ void TextInputHandler::InsertText(NSString* aString,
   }
 
   // XXX Shouldn't we hold mDispatcher instead of mWidget?
-  RefPtr<nsCocoaWindow> widget(mWidget);
+  RefPtr<nsChildView> widget(mWidget);
   nsresult rv = mDispatcher->BeginNativeInputTransaction();
   if (NS_WARN_IF(NS_FAILED(rv))) {
     MOZ_LOG_KEY_OR_IME(LogLevel::Error,
@@ -2608,16 +2609,8 @@ void TextInputHandler::InsertText(NSString* aString,
   // If mCurrentKeyEvent.mKeyEvent is null, the text should be inputted as
   // composition events.
   nsEventStatus status = nsEventStatus_eIgnore;
-  bool keyPressDispatched = [&]() {
-    // If text content is chrome process, OnTextChange etc will be dispatched
-    // synchronously. We don't want to dismiss text substitution panel at this
-    // point.
-    AutoRestore<bool> block(mBlockDismissTextSubstitutionPanel);
-    mBlockDismissTextSubstitutionPanel = true;
-
-    return mDispatcher->MaybeDispatchKeypressEvents(keypressEvent, status,
-                                                    currentKeyEvent);
-  }();
+  bool keyPressDispatched = mDispatcher->MaybeDispatchKeypressEvents(
+      keypressEvent, status, currentKeyEvent);
   bool keyPressHandled = (status == nsEventStatus_eConsumeNoDefault);
 
   // WebKit and text editor dismisses autocorrect panel by space, then process
@@ -2750,7 +2743,7 @@ bool TextInputHandler::HandleCommand(Command aCommand) {
     }
   }
 
-  RefPtr<nsCocoaWindow> widget(mWidget);
+  RefPtr<nsChildView> widget(mWidget);
   nsresult rv = mDispatcher->BeginNativeInputTransaction();
   if (NS_WARN_IF(NS_FAILED(rv))) {
     MOZ_LOG_KEY_OR_IME(LogLevel::Error,
@@ -3059,7 +3052,7 @@ bool TextInputHandler::HandleCommand(Command aCommand) {
 }
 
 bool TextInputHandler::DoCommandBySelector(const char* aSelector) {
-  RefPtr<nsCocoaWindow> widget(mWidget);
+  RefPtr<nsChildView> widget(mWidget);
 
   KeyEventState* currentKeyEvent = GetCurrentKeyEvent();
 
@@ -3120,17 +3113,9 @@ bool TextInputHandler::DoCommandBySelector(const char* aSelector) {
     currentKeyEvent->InitKeyEvent(this, keypressEvent, false);
 
     nsEventStatus status = nsEventStatus_eIgnore;
-    // If text content is chrome process, OnTextChange etc will be dispatched
-    // synchronously. We don't want to dismiss text substitution panel at this
-    // point.
-    {
-      AutoRestore<bool> block(mBlockDismissTextSubstitutionPanel);
-      mBlockDismissTextSubstitutionPanel = true;
-
-      currentKeyEvent->mKeyPressDispatched =
-          mDispatcher->MaybeDispatchKeypressEvents(keypressEvent, status,
-                                                   currentKeyEvent);
-    }
+    currentKeyEvent->mKeyPressDispatched =
+        mDispatcher->MaybeDispatchKeypressEvents(keypressEvent, status,
+                                                 currentKeyEvent);
     currentKeyEvent->mKeyPressHandled =
         (status == nsEventStatus_eConsumeNoDefault);
     MOZ_LOG_KEY_OR_IME(
@@ -3993,7 +3978,7 @@ bool IMEInputHandler::MaybeDispatchCurrentKeydownEvent(bool aIsProcessedByIME) {
   // Mark currentKeyEvent as "dispatched eKeyDown event" and actually do it.
   currentKeyEvent->mKeyDownDispatched = true;
 
-  RefPtr<nsCocoaWindow> widget(mWidget);
+  RefPtr<nsChildView> widget(mWidget);
 
   WidgetKeyboardEvent keydownEvent(true, eKeyDown, widget);
   // Don't mark the eKeyDown event as "processed by IME" if the composition
@@ -4651,7 +4636,7 @@ NSArray* IMEInputHandler::GetValidAttributesForMarkedText() {
  *
  ******************************************************************************/
 
-IMEInputHandler::IMEInputHandler(nsCocoaWindow* aWidget,
+IMEInputHandler::IMEInputHandler(nsChildView* aWidget,
                                  NSView<mozView>* aNativeView)
     : TextInputHandlerBase(aWidget, aNativeView),
       mPendingMethods(0),
@@ -4717,7 +4702,7 @@ void IMEInputHandler::OnFocusChangeInGecko(bool aFocus) {
   ResetTimer();
 }
 
-bool IMEInputHandler::OnDestroyWidget(nsCocoaWindow* aDestroyingWidget) {
+bool IMEInputHandler::OnDestroyWidget(nsChildView* aDestroyingWidget) {
   MOZ_LOG(gIMELog, LogLevel::Info,
           ("%p IMEInputHandler::OnDestroyWidget, aDestroyingWidget=%p, "
            "sFocusedIMEHandler=%p, IsIMEComposing()=%s",
@@ -4725,7 +4710,7 @@ bool IMEInputHandler::OnDestroyWidget(nsCocoaWindow* aDestroyingWidget) {
            TrueOrFalse(IsIMEComposing())));
 
   // If we're not focused, the focused IMEInputHandler may have been
-  // created by another widget/nsCocoaWindow.
+  // created by another widget/nsChildView.
   if (sFocusedIMEHandler && sFocusedIMEHandler != this) {
     sFocusedIMEHandler->OnDestroyWidget(aDestroyingWidget);
   }
@@ -5038,10 +5023,8 @@ void IMEInputHandler::HandleTextSubstitution(
   }
 
   // Dismiss text substitution panel since this text change might be script etc
-  if (!mBlockDismissTextSubstitutionPanel) {
-    mProcessTextSubstitution = false;
-    DismissTextSubstitutionPanel();
-  }
+  mProcessTextSubstitution = false;
+  DismissTextSubstitutionPanel();
 
   // Set new text substitution data to show its panel by current changed text.
   //
@@ -5171,10 +5154,16 @@ void IMEInputHandler::OnTextSubstitution(uint32_t aStartOffset) {
   // NSTextCheckingResult.range is read only, so re-create this result object.
   NSRange candidatedRange = NSMakeRange(candidate.range.location + startFetch,
                                         candidate.range.length);
+  NSArray<NSString *> *anAlternativeString;
+  if(@available(macOS 10.8, *)) {
+    anAlternativeString = candidate.alternativeStrings;
+  } else {
+    anAlternativeString = @[];
+  }
   mCandidatedTextSubstitutionResult = [[NSTextCheckingResult
       correctionCheckingResultWithRange:candidatedRange
                       replacementString:candidate.replacementString
-                     alternativeStrings:candidate.alternativeStrings] retain];
+                     alternativeStrings:anAlternativeString] retain];
   mOriginalTextForTextSubstitution =
       Substring(queryTextContentEvent.mReply->DataRef(),
                 candidatedRange.location - startFetch, candidatedRange.length);
@@ -5224,12 +5213,17 @@ void IMEInputHandler::ShowTextSubstitutionPanel() {
   if (!spellchecker) {
     return;
   }
+  NSArray<NSString *> *anotherAlternativeString;
+  if(@available(macOS 10.8, *)) {
+    anotherAlternativeString = mCandidatedTextSubstitutionResult.alternativeStrings;
+  } else {
+    anotherAlternativeString = @[];
+  }
   [spellchecker
       showCorrectionIndicatorOfType:NSCorrectionIndicatorTypeDefault
                       primaryString:mCandidatedTextSubstitutionResult
                                         .replacementString
-                 alternativeStrings:mCandidatedTextSubstitutionResult
-                                        .alternativeStrings
+                 alternativeStrings:anotherAlternativeString
                     forStringInRect:rect
                                view:mView
                   completionHandler:^(NSString* aAcceptedString) {
@@ -5273,11 +5267,6 @@ void IMEInputHandler::ShowTextSubstitutionPanel() {
 void IMEInputHandler::DismissTextSubstitutionPanel() {
   NS_OBJC_BEGIN_TRY_IGNORE_BLOCK;
 
-  MOZ_LOG(gIMELog, LogLevel::Info,
-          ("%p IMEInputHandler::DismissTextSubstitutionPanel, "
-           "mProcessTextSubstitution=%s",
-           this, mProcessTextSubstitution ? "true" : "false"));
-
   NSSpellChecker* spellchecker = [NSSpellChecker sharedSpellChecker];
   if (!spellchecker) {
     return;
@@ -5325,7 +5314,7 @@ bool IMEInputHandler::OnHandleEvent(NSEvent* aEvent) {
   }
 
   bool allowConsumeEvent = true;
-  if (!IsIMEComposing()) {
+  if (nsCocoaFeatures::OnCatalinaOrLater() && !IsIMEComposing()) {
     // Hack for bug of Korean IMEs on Catalina (10.15).
     // If we are inactivated during composition, active Korean IME keeps
     // consuming all mousedown events of any mouse buttons.  So, we should
@@ -5361,7 +5350,7 @@ int32_t TextInputHandlerBase::sSecureEventInputCount = 0;
 NS_IMPL_ISUPPORTS(TextInputHandlerBase, TextEventDispatcherListener,
                   nsISupportsWeakReference)
 
-TextInputHandlerBase::TextInputHandlerBase(nsCocoaWindow* aWidget,
+TextInputHandlerBase::TextInputHandlerBase(nsChildView* aWidget,
                                            NSView<mozView>* aNativeView)
     : mWidget(aWidget), mDispatcher(aWidget->GetTextEventDispatcher()) {
   gHandlerInstanceCount++;
@@ -5375,7 +5364,7 @@ TextInputHandlerBase::~TextInputHandlerBase() {
   }
 }
 
-bool TextInputHandlerBase::OnDestroyWidget(nsCocoaWindow* aDestroyingWidget) {
+bool TextInputHandlerBase::OnDestroyWidget(nsChildView* aDestroyingWidget) {
   MOZ_LOG_KEY_OR_IME(LogLevel::Info,
                      ("%p TextInputHandlerBase::OnDestroyWidget, "
                       "aDestroyingWidget=%p, mWidget=%p",
@@ -5430,7 +5419,7 @@ nsresult TextInputHandlerBase::SynthesizeNativeKeyEvent(
                     modifierFlags:modifierFlags
                         timestamp:0
                      windowNumber:windowNumber
-                          context:nil
+                          context:[NSGraphicsContext currentContext]
                        characters:XPCOMStringToNSString(aCharacters)
       charactersIgnoringModifiers:XPCOMStringToNSString(aUnmodifiedCharacters)
                         isARepeat:NO
@@ -5696,7 +5685,7 @@ void TextInputHandlerBase::KeyEventState::InitKeyEvent(
                           modifierFlags:[mKeyEvent modifierFlags]
                               timestamp:[mKeyEvent timestamp]
                            windowNumber:[mKeyEvent windowNumber]
-                                context:nil
+                                context:[mKeyEvent context]
                              characters:unhandledNSString
             charactersIgnoringModifiers:[mKeyEvent charactersIgnoringModifiers]
                               isARepeat:[mKeyEvent isARepeat]
-- 
2.53.0

