From dbb61b53f1df4e405dd17ce7a055efb37a81cf90 Mon Sep 17 00:00:00 2001
From: aobaharuki2005 <tranbaohnth@outlook.com>
Date: Sun, 19 Apr 2026 22:37:08 +0700
Subject: [PATCH 213/246] merge_widget/nsBaseWidget.h

---
 widget/nsBaseWidget.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/widget/nsBaseWidget.h b/widget/nsBaseWidget.h
index d10aa44db6..1add7b1f2f 100644
--- a/widget/nsBaseWidget.h
+++ b/widget/nsBaseWidget.h
@@ -359,6 +359,12 @@ class nsBaseWidget : public nsIWidget, public nsSupportsWeakReference {
   bool IsSmallPopup() const;
 
   PopupLevel GetPopupLevel() { return mPopupLevel; }
+  
+  LayoutDeviceIntSize ClientToWindowSize(
+      const LayoutDeviceIntSize& aClientSize) override {
+    return aClientSize;
+  }
+
 
   const SizeConstraints GetSizeConstraints() override;
   void SetSizeConstraints(const SizeConstraints& aConstraints) override;
-- 
2.53.0

